src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParserStaticMethods.cs

```diff
@@ -103,62 +111,92 @@ internal static byte[] ObfuscatePassword(string password)
+                bHi = (byte)(password[i] & 0xf0);
+                password[i] = (byte)(((bHi >> 4) | (bLo << 4)) ^ 0xa5);
+            }
+            return password;
         }
 
+        private const int NoProcessId = -1;
+        private static int s_currentProcessId = NoProcessId;
+        internal static int GetCurrentProcessIdForTdsLoginOnly()
+        {
+            if (s_currentProcessId == NoProcessId)
+            {
+                // Pick up the process Id from the current process instead of randomly generating it.
+                // This would be helpful while tracing application related issues.
+                int processId;
+                using (System.Diagnostics.Process p = System.Diagnostics.Process.GetCurrentProcess())
+                {
+                    processId = p.Id;
+                }
+                System.Threading.Volatile.Write(ref s_currentProcessId, processId);
+            }
+            return s_currentProcessId;
+        }
 
-        [SecurityPermission(SecurityAction.Assert, Flags = SecurityPermissionFlag.UnmanagedCode)]
-        [ResourceExposure(ResourceScope.None)] // SxS: we use this method for TDS login only
-        [ResourceConsumption(ResourceScope.Process, ResourceScope.Process)]
-        static internal Int32 GetCurrentThreadIdForTdsLoginOnly()
+
+        internal static int GetCurrentThreadIdForTdsLoginOnly()
         {
-#pragma warning disable 618
-            return AppDomain.GetCurrentThreadId(); // don't need this to be support fibres;
-#pragma warning restore 618
+            return Environment.CurrentManagedThreadId;
         }
 
+        private static byte[] s_nicAddress = null;
 
         [ResourceExposure(ResourceScope.None)] // SxS: we use MAC address for TDS login only
         [ResourceConsumption(ResourceScope.Machine, ResourceScope.Machine)]
         static internal byte[] GetNetworkPhysicalAddressForTdsLoginOnly()
         {
-            // NIC address is stored in NetworkAddress key.  However, if NetworkAddressLocal key
-            // has a value that is not zero, then we cannot use the NetworkAddress key and must
-            // instead generate a random one.  I do not fully understand why, this is simply what
-            // the native providers do.  As for generation, I use a random number generator, which
-            // means that different processes on the same machine will have different NIC address
-            // values on the server.  It is not ideal, but native does not have the same value for
-            // different processes either.
-
-            const string key = "NetworkAddress";
-            const string localKey = "NetworkAddressLocal";
-            const string folder = "SOFTWARE\\Description\\Microsoft\\Rpc\\UuidTemporaryData";
-
-            int result = 0;
-            byte[] nicAddress = null;
-
-            object temp = ADP.LocalMachineRegistryValue(folder, localKey);
-            if (temp is int)
+            if (s_nicAddress != null)
```

Comment:
I chose a merger of the netfx and netcore implementations of GetNetworkPhysicalAddressForTdsLoginOnly. Netcore had the optimization of caching s_nicAddress but did not have the registry lookup when running on Windows. Since this introduces registry lookups when on Windows, we can bring that functionality to netcore since users generally expect behavior parity when migrating from netfx to netcore.